core/src/main/java/nl/nn/adapterframework/doc/model/ConfigChildSet.java

```diff
@@ -0,0 +1,139 @@
+package nl.nn.adapterframework.doc.model;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import lombok.Getter;
+import lombok.Setter;
+
+/**
+ * Holds the list of all cumulative config children sharing some role name, say R, but
+ * only if there is no ancestor having the same config children for role name R.
+ * <p>
+ * Example. Assume that a class Super has a derived class Derived. Super has a method
+ * registerChild(A a) while Derived has a method registerChild(B b). Then the cumulative
+ * config children of Derived with role name "child" are (A, child), (B, child). These
+ * two go into a ConfigChildSet with common role name "child".
+ * <p>
+ * If the method registerChild(B b) would not be in Derived but in Super, then the cumulative
+ * config children with role name "child" would be the same for Derived and Super. Therefore
+ * only Super would have a ConfigChildSet for this role name, not Derived.
+ * @author martijn
+ *
+ */
+public class ConfigChildSet {
+	private final @Getter List<ConfigChild> configChildren;
+	private @Getter @Setter ElementRoleSet elementRoleSet;
+
+	/**
+	 * @throws IllegalArgumentException if input children is empty or when the elements
+	 * are not in the right sort order
+	 * or do not share a role name. The children should be sorted from derived to ancestor
+	 * owning element and then by order. This precondition should be enforced
+	 * by {@link AncestorChildNavigation}.
+	 */
+	ConfigChildSet(List<ConfigChild> configChildren) {
+		this.configChildren = configChildren;
+		if(configChildren.isEmpty()) {
+			throw new IllegalArgumentException("A config child cannot have an empty list of config childs");
+		}
+		if(configChildren.size() >= 2) {
+			FrankElement parent = configChildren.get(0).getOwningElement();
+			int order = configChildren.get(0).getOrder();
+			for(ConfigChild c: configChildren.subList(1, configChildren.size())) {
+				if(c.getOwningElement() != parent) {
+					if(c.getOwningElement() != parent.getNextAncestorThatHasConfigChildren(ElementChild.ALL)) {
+						throw new IllegalArgumentException(String.format("Cumulative config children are not sorted: [%s] should not be followed by [%s]",
+								parent.getFullName(), c.getOwningElement().getFullName()));
+					}
+					parent = c.getOwningElement();
+				}
+				else {
+					if(! (order <= c.getOrder())) {
+						throw new IllegalArgumentException(String.format("Cumulative config children are not sorted by order. Offending config child [%s]",
```

Comment:
Ik zie dit meer als een `IllegalStateException`

Reply:
Er zitten meerdere throw statements in deze constructor. Wil je bij allemaal IllegalStateException, of is voor sommige andere throw statements IllegalArgumentException wel goed?

Reply:
Ik heb voor alle throw statements hier de exception veranderd